### PR TITLE
Lazy launching BrowseEnv / making BrowseEnv optional

### PR DESCRIPTION
**Pull request by assertion**
Details are here: https://github.com/OpenDevin/OpenDevin/issues/2120

### CODE REVIEW

File: opendevin/runtime/browser/browser_env.py [hunk 1]
```diff
@@ -32,12 +33,18 @@ def __init__(self):
         self.process = multiprocessing.Process(
             target=self.browser_process,
         )
+        if is_async:
+            threading.Thread(target=self.init_browser).start()
+        else:
+            self.init_browser()
+        atexit.register(self.close)
+
+    def init_browser(self):
```
Comment (yufansong) [inline opendevin/runtime/browser/browser_env.py:42]: Do we want to impelement the lazy luanching or async launching?

Comment (neubig) [reply opendevin/runtime/browser/browser_env.py:42]: I would argue that we could add that to a different PR if necessary, this is already an improvement.

**Review by neubig — APPROVED** [review]
I tested and looks great, approved!

File: agenthub/codeact_swe_agent/codeact_swe_agent.py [hunk 2]
```diff
@@ -127,6 +128,7 @@ class CodeActSWEAgent(Agent):
         AgentSkillsRequirement(),
         JupyterRequirement(),
     ]
+    runtime_tools: list[RuntimeTool] = [RuntimeTool.BROWSER]
```
Comment (li-boxuan) [inline agenthub/codeact_swe_agent/codeact_swe_agent.py:131]: I don't think CodeActSWEAgent needs browsing

Comment (li-boxuan) [reply agenthub/codeact_swe_agent/codeact_swe_agent.py:131]: never mind let's do it in a separate PR. CodeActSWEAgent needs some polish.

File: opendevin/runtime/browser/browser_env.py (hunk 1 above)
Comment (li-boxuan) [reply opendevin/runtime/browser/browser_env.py:42]: Looks like this PR doesn't only make it optional (lazy launching), but also make the launching itself async. Two improvements.

**Pull request merged**